### PR TITLE
feat(responsemanager): allow ctx augmentation via queued request hook

### PR DESCRIPTION
**Pull request by rvagg**
Closes: https://github.com/ipfs/go-graphsync/issues/297

### CODE REVIEW

File: responsemanager/responsemanager_test.go [hunk 1]
```diff
@@ -64,11 +77,20 @@ func TestIncomingQuery(t *testing.T) {
...
 	// ensure request queued hook fires.
 	out := <-qhc
 	require.Equal(t, td.p, out.p)
 	require.Equal(t, out.request.ID(), td.requestID)
 	td.connManager.RefuteProtected(t, td.p)
+
+	tracing := collectTracing(t)
+	require.ElementsMatch(t, []string{
+		"TestIncomingQuery(0)->response(0)->executeTask(0)",
+	}, tracing.TracesToStrings())
```
Comment (rvagg) [inline responsemanager/responsemanager_test.go:93]: here we demonstrate the ability to insert an arbitrary parent to the `response` trace

File: responsemanager/hooks/requesthook.go [hunk 2]
```diff
@@ -24,12 +25,14 @@ type IncomingRequestQueuedHooks struct {
 type internalRequestQueuedHookEvent struct {
 	p       peer.ID
 	request graphsync.RequestData
+	reqCtx  context.Context
```
Comment (hannahhoward) [inline responsemanager/hooks/requesthook.go:28]: do you actually need this context here?

File: responsemanager/hooks/requesthook.go [hunk 3]
```diff
@@ -82,6 +87,10 @@ func (irh *IncomingRequestHooks) Register(hook graphsync.OnIncomingRequestHook)
 	return graphsync.UnregisterHookFunc(irh.pubSub.Subscribe(hook))
 }
 
+type QueuedRequestResult struct {
```
Comment (hannahhoward) [inline responsemanager/hooks/requesthook.go:90]: It's also ok to just remove this struct and return just the new context if that is all that's needed

File: responsemanager/hooks/requesthook.go (hunk 2 above)
Comment (hannahhoward) [reply responsemanager/hooks/requesthook.go:28]: I'm pretty sure it's ultimately unused

**Review by hannahhoward — CHANGES_REQUESTED** [review]
Unless I'm wrong remove the unneeded struct member. Otherwise LGTM

File: responsemanager/hooks/requesthook.go (hunk 2 above)
Comment (rvagg) [reply responsemanager/hooks/requesthook.go:28]: good spotting

File: responsemanager/hooks/requesthook.go (hunk 3 above)
Comment (rvagg) [reply responsemanager/hooks/requesthook.go]: :thumbsup:

**Review by hannahhoward — APPROVED** [review]

**Pull request merged**